idmapping-rest/src/test/java/org/uniprot/api/idmapping/service/TestConfig.java

```diff
@@ -18,29 +11,34 @@
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Profile;
 import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
 import org.uniprot.api.idmapping.service.impl.RedisCacheMappingJobService;
 
-import redis.embedded.RedisServer;
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * @author sahmad
  * @created 24/02/2021
  */
 @TestConfiguration
+@Testcontainers
 public class TestConfig {
-    private final RedisServer redisServer;
-
-    public TestConfig() {
-        this.redisServer = new RedisServer(6379);
-    }
+    public static final String REDIS_IMAGE_VERSION = "redis:5.0.3-alpine";
+    @Container
+    public GenericContainer redisServer =
+            new GenericContainer(DockerImageName.parse(REDIS_IMAGE_VERSION))
+                    .withExposedPorts(6379)
+                    .withReuse(true);
 
     @PostConstruct
```

Comment:
Please remove @PostConstruct and @PreDestroy methods. @Container annotation should take care of docker image as and when needed by test. It will also stop the container once the test is done.